### PR TITLE
Simple Forms API: Increase Faraday timeout and add logging

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary

This PR is an initial attempt to deal with the occasional `Gateway Time-out` error that we've been seeing around PDF form submission. It does two things:

- Increase `self.read_timeout` for our Faraday client from `20` to `30`.
- Add a line of logging just before we attempt to upload the document.

The first thing _might_ just solve the problem (but I'm not counting on it). If it does not, the next step would be to use `in_parallel` or something async with Faraday to handle the fact that uploading docs is blocking. However, there is hope that simply increasing the timeout will suffice since I don't think we have high traffic and it may just be that these PDFs are too big/take too long to upload. An async worker may or may not be the actual solution.

The second thing adds a line of logging so that we can get a better picture in the logs of how this is failing, since the stack trace _looks_ kind of incomplete to me.

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/513

## Testing done

None. This is hard to test. If anyone has any tips of how to test this I am all ears!!

## What areas of the site does it impact?
Simple forms api

## Requested Feedback
This is an exploratory PR since I don't feel very confident of how to actually solve the problem so if anyone has better ideas, I'd love to hear 'em!

### CODE REVIEW

**Review by LindseySaari — COMMENTED** [review]
Which downstream service does this send the request to? Have you coordinated with them that this timeout is an acceptable change?

**Comment by Thrillberg** [comment]
@LindseySaari Thanks for taking a look at this! The Lighthouse Benefits Intake API Service receives uploads from the `SimpleFormsApiSubmission` service. I have not coordinated with them, though. Are you able to point me in the right direction? I looked at the docs but didn't see who to contact with questions.

**Comment by patrickvinograd** [comment]
As of last time I was aware, Lighthouse Benefits Intake sends to the "Central Mail" upstream service

**Comment by michelpmcdonald** [comment]
> As of last time I was aware, Lighthouse Benefits Intake sends to the "Central Mail" upstream service

Yep, we do, the upload to central mail is async and not really part of the upload call cycle ya'll make.

The "location" url that we provide in the initial /upload call is actually a url to an amazon s3 bucket.

Once the upload to s3 is complete, our benefits intake code gets a webhook callback from s3.  At that point, we create a new sidekiq job to grab the file from s3 and upload the file to Central Mail.  The sidekiq job is configured with multiple retries so if something goes wrong we re-attempt the central mail upload.....

i might of been the one that suggested ya'll consider using benefits intake to take advantage of our retries n such instead of ~(re)creating that same logic in the simple forms code base.

**Comment by patrickvinograd** [comment]
So this raises the question - why is this timing out, if it's an upload from vets-api to the benefits intake API (which is really an upload to an S3 bucket in the same cloud environment and should be plenty fast)?

Is it possible that the (upload from client to vets-api) and (upload from vets-api to BenefitsIntake) are coupled, and that a slow client is the issue here? If so, then sure, bumping the timeout of the BenefitsIntake connection will help, but is also kind of kicking the can down the road as there's no reasonable bound you can define on client bandwidth/upload speed. 

Which is why the BenefitsIntake API was built the way it was with a direct signed upload to S3 - to minimize the impact of slow clients on vets-api;  which you've now re-introduced 🤔

**Comment by Thrillberg** [comment]
Thanks for your thoughts on this @michelpmcdonald and @patrickvinograd! The call to the Benefits Intake API is decoupled from any client interaction. The PDF that is being uploaded via Lighthouse is [generated by `vets-api`](https://github.com/department-of-veterans-affairs/vets-api/blob/1bf9b62dfaeb8f1c05e9b6ad24b87dc41f9de939/modules/simple_forms_api/app/controllers/simple_forms_api/v1/uploads_controller.rb#L29). I don't really understand the implications of why it being an S3 buckets means it'll be plenty fast, but we're getting timeouts anyway.

Also, @beaugrantham-va has indicated to me on Slack that the Lighthouse API enforces a 20s timeout so my change here wouldn't actually make a difference anyway. I think I'm back to square one in debugging this.

I'll close this PR here since we know we _don't_ want the 30s change but I'd love to keep the conversation going, either here or on Slack, since I'm not sure where else to turn in terms of debugging this. Thanks everyone!

**Comment by patrickvinograd** [comment]
It's not the S3 that should make it fast, it's the fact that once uploaded from the client, all the relevant infrastructure is in the same AWS environment (gov-cloud-west) - so I wouldn't expect any meaningful network latency that would account for a slowdown. Whereas you can't assume the same for the initial upload from the user's device out on the internet. 

But since you say the two hops are not coupled, there must be some source of latency. I think the path forward is to really isolate where that latency is coming from with better instrumentation, before we go adjusting timeouts in hope that will make the problem go away.

**Comment by Thrillberg** [comment]
@patrickvinograd I've circled back with @michelpmcdonald and he and @kristen-brown suggest trying to increase the time out such as I was doing in this PR because my calls from simple forms are never even making it to their code. What do you think of merging this code and then seeing if it makes a difference? I think otherwise we're not sure where else to investigate. If we really want to take it slower, we can merge only the additional logging part of this PR but hopefully it's harmless to increase the timeout? Let me know what you think!

**Review by michelpmcdonald — APPROVED** [review]
Yep, looks GTM

**Comment by michelpmcdonald** [comment]
Related slack thread
https://lighthouseva.slack.com/archives/C010ENX8X7G/p1694532020506159

tl;dr we are seeing occasional timeouts with endpoint in Benefits Intake and Forms(not Simple Forms) API
